compiler/prelude/prelude.go

```diff
@@ -564,4 +564,37 @@ var $unsafeMethodToFunction = function(typ, name, isPtr) {
      }
   }
 };
+
+var $id = function(x) {
+  return x;
+};
+
+var $instanceOf = function(x, y) {
+  return x instanceof y;
+};
+
+var $getValueType = function(x) {
+  if (typeof(x) === "undefined") {
+    return 0; // TypeUndefined
+  }
+  if (x === null) {
+    return 1; // TypeNull
+  }
+  if (typeof(x) === "boolean") {
+    return 2; // TypeBoolean
+  }
+  if (typeof(x) === "number") {
+    return 3; // TypeNumber
+  }
+  if (typeof(x) === "string") {
+    return 4; // TypeString
+  }
+  if (typeof(x) === "symbol") {
+    return 5; // TypeSymbol
+  }
+  if (typeof(x) === "function") {
+    return 7; // TypeFunction
+  }
+  return 6; // TypeObject
+};
```

Comment:
Since the mapping between a string type and a number is a `syscall/js`-specific logic, I would only define a `$typeOf` helper in prelude and implement the rest as a Go function in the `syscall/js` package. Then you'd be able able to use Go constants instead of numbers.

Reply:
Okay @nevkontakte I'll do that, thanks for looking.

Reply:
Okay @nevkontakte I finished the rest as a Go function, let me know if there's anything else I should change and I'll tackle it Wednesday morning CDT. Thanks for helping me through this.